napari/layers/labels/labels.py

```diff
@@ -174,11 +174,24 @@ def __init__(
         self._background_label = 0
         self._num_colors = num_colors
         self._random_colormap = label_colormap(self.num_colors)
+        self._all_vals = low_discrepancy_image(
+            np.arange(self.num_colors), self._seed
+        )
+        self._all_vals[0] = 0
         self._color_mode = LabelColorMode.AUTO
         self._brush_shape = LabelBrushShape.CIRCLE
         self._show_selected_label = False
         self._contour = 0
 
+        data = [data] if not isinstance(data, list) else data
+        if any(np.issubdtype(d.dtype, np.floating) for d in data):
+            warnings.warn(
+                "Float dtypes are not supported for Labels layers. Converting data to integers..."
+            )
+            data = [d.astype(np.int32) for d in data]
+
+        data = data[0] if len(data) == 1 else data
```

Comment:
This is a slight replication of our `guess_multiscale` code, so I guess here we should use that... And we might want to use int64 rather than int32 since the max integer value for float64 is 2**53.

```python
looks_multiscale, data = guess_multiscale(data)
if not looks_multiscale:
    data = [data]
if any(np.issubdtype(d.dtype, np.floating) for d in data):
    warnings.warn()
    int_data = []
    for d in data:
        # convert float values to corresponding integers
        if d.dtype == np.float64:
            int_data.append(d.astype(np.int64))
        elif d.dtype == np.float32:
            int_data.append(d.astype(np.int32))
        else: 
            int_data.append(d)
    data = int_data
if not looks_multiscale:
    data = data[0]
```

This code is (I think) more robust but longer, so maybe it should be pulled out into a utility function:

```python
data = ensure_int_labels(data)
```